src/menu.tsx

```diff
@@ -1,18 +1,159 @@
 import * as React from 'react';
 import classnames from 'classnames';
-import {ESC, TAB} from './utils/key-codes';
+import {ESC, LEFT, RIGHT, UP, DOWN, ENTER, SPACE, TAB} from './utils/key-codes';
 import {cancelEvent, getPrefixedDataAttributes} from './utils/dom';
 import Overlay from './overlay';
 import * as styles from './menu.css';
 import {useWindowSize} from './hooks';
 import {Portal} from './portal';
 import {assignInlineVars} from '@vanilla-extract/dynamic';
+import Box from './box';
+import Inline from './inline';
+import Touchable from './touchable';
+import {Text3} from './text';
+import {vars} from './skins/skin-contract.css';
+import Divider from './divider';
+import Checkbox from './checkbox';
+import {CSSTransition} from 'react-transition-group';
+import {combineRefs} from './utils/common';
 
-import type {DataAttributes} from './utils/types';
+import type {DataAttributes, IconProps} from './utils/types';
+
+const MENU_TRANSITION_DURATION_IN_MS = 120;
+
+type MenuContextType = {
+    focusedValue: string | null;
+    isMenuOpen: boolean;
+    setFocusedValue: (value: string | null) => void;
+    closeMenu: () => void;
+};
+const MenuContext = React.createContext<MenuContextType>({
+    focusedValue: null,
+    isMenuOpen: false,
+    setFocusedValue: () => {},
+    closeMenu: () => {},
+});
+export const useMenuContext = (): MenuContextType => React.useContext(MenuContext);
+
+interface MenuItemBaseProps {
+    label: string;
+    Icon?: React.FC<IconProps>;
+    destructive?: boolean;
+    disabled?: boolean;
+    'aria-label'?: string;
+    children?: undefined;
+    onPress: (label: string) => void;
+}
+
+interface MenuItemWithoutControlProps extends MenuItemBaseProps {
+    controlType?: undefined;
+    checked?: undefined;
+}
+
+interface MenuItemWithControlProps extends MenuItemBaseProps {
+    controlType: 'checkbox';
+    checked?: boolean;
+}
+
+type MenuItemProps = MenuItemWithControlProps | MenuItemWithoutControlProps;
+
+export const MenuItem: React.FC<MenuItemProps> = ({
+    label,
+    Icon,
+    destructive,
+    disabled,
+    'aria-label': ariaLabel,
```

Comment:
does it make sense to have an aria-label != label ? do you have any example?

Reply:
This was a request from vivo. They want to be able to choose their own aria labels

Reply:
removed aria-label as prop